python/src/trezorlib/cli/firmware.py

```diff
@@ -357,52 +351,43 @@ def cli():
 
 @cli.command()
 # fmt: off
-@click.option("-f", "--filename", type=click.File("rb"), help="File containing firmware data")
-@click.option("-s", "--skip-device-check", is_flag=True, help="Do not validate the device")
+@click.argument("filename", type=click.File("rb"))
+@click.option("-c", "--check-device", is_flag=True, help="Validate the device compatibility")
 @click.option("--fingerprint", help="Expected firmware fingerprint in hex")
 # fmt: on
-@with_client
 def verify(
-    client: TrezorClient,
     filename: BinaryIO,
-    skip_device_check: bool,
+    check_device: bool,
     fingerprint: str,
 ) -> None:
-    """Verifying the integrity of the firmware data stored in a file.
+    """Verify the integrity of the firmware data stored in a file.
 
-    By default validates both the fingerprint (if specified) and the device
-        integrity
+    By default the device is not checked and does not need to be connected.
+    Its validation must be specified.
 
-    Has 1 as the exit code for all the issues (catching all problem's exit codes)
+    In case of validation failure exits with the appropriate exit code.
     """
-    if not filename:
-        click.echo("Please include the filename by '-f' or '--filename'.")
-        sys.exit(1)
-
-    # Deciding if to validate device integrity
-    if skip_device_check:
+    # Deciding if to take the device into account
+    if check_device:
+        client = get_default_client()
```

Comment:
this way the client won't respect the passed in `--path` parameter and possibly some other things.

instead, please use `obj.client_context()`, see `trezorctl.py::get_session` for a correct example